src/query/service/src/pipelines/processors/transforms/hash_join/result_blocks.rs

```diff
@@ -270,6 +281,118 @@ impl JoinHashTable {
         Ok(())
     }
 
+    fn create_marker_block(&self, has_null: bool, markers: Vec<MarkerKind>) -> Result<DataBlock> {
```

Comment:
It seems the method is duplicated with `mark_join_blocks`, better to extract the same logic to one method.

Reply:
ok

Reply:
refactor done